### ISSUE TITLE
`Deps` type requires an array, although this is useless

### ISSUE THREAD

**Issue by Fyzu** [artalar/usePureCallback#1]
Dependencies now require them to be an array `Deps extends any[]`.

But this is useless because we pass them as is, `fn(argsRef.current.deps, ...args)`.

**Comment by artalar** [comment]
You mean we could write `usePureFunction({some: 42}, ({some}) => ...)` and it still will work? I agree, but for consistency strict array of dependencies looks better, IMHO.

**Comment by Fyzu** [comment]
Yep, or simpler: `usePureCallback(onChange, (onChange, id) => onChange(id))`.

To me that if this isn't part of the logic, you shouldn't force the code-style things.
